### ISSUE TITLE
Resolve flaky IDEA Inspections

### ISSUE THREAD

**Issue by nrmancuso** [checkstyle/checkstyle#12359]
We can use this issue to track flaky inspections.

Noticed at https://github.com/checkstyle/checkstyle/pull/12354 and https://github.com/checkstyle/checkstyle/pull/12357

On commit https://github.com/checkstyle/checkstyle/commit/73d92727546cb417ad630359263e05e7e434317e:
```
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java b/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java
index e97a0df1c18..eb099f3ce2a 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java
@@ -97,8 +97,6 @@ public interface DetailAST {
      *      Usage of this method is no longer accepted. We encourage
      *      traversal of subtrees to be written per the needs of each check
      *      to avoid unintended side effects.
-     * @noinspection DeprecatedIsStillUsed
-     * @noinspectionreason DeprecatedIsStillUsed - Method used in unit testing
      */
     @Deprecated(since = "8.43")
     boolean branchContains(int type);

```
We have failing inspection: `DeprecatedIsStillUsed`. But, when we have `noinspection` tag in place, inspections reports `RedundantSuppression`. We used hack at https://github.com/checkstyle/checkstyle/pull/12357/commits/e27254ed879c031f2145110bf575feefd15aad1a to suppress false positive of `RedundantSuppression`:
```
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java b/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java
index e97a0df1c18..56a505f3b24 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java
@@ -97,8 +97,9 @@ public interface DetailAST {
      *      Usage of this method is no longer accepted. We encourage
      *      traversal of subtrees to be written per the needs of each check
      *      to avoid unintended side effects.
-     * @noinspection DeprecatedIsStillUsed
+     * @noinspection DeprecatedIsStillUsed, RedundantSuppression
      * @noinspectionreason DeprecatedIsStillUsed - Method used in unit testing
+     * @noinspectionreason RedundantSuppression - Inspections shows false positive
      */
     @Deprecated(since = "8.43")
     boolean branchContains(int type);

```

We should open an issue with JetBrains for this.
